### PR TITLE
CC-2594 Refactor GetServiceEndpoints

### PR DESCRIPTION
**Pull request by ed-**
Removed GetServiceEndpoints from node.
Removed several helper functions from node as a result.
Removed node/agent_proxy_test.go because it was testing only these helpers.

Created new createControlplaneEndpoints in controller, and test in container/controller_test.go as a result.

### CODE REVIEW

File: container/controller.go [hunk 1]
```diff
@@ -790,103 +790,106 @@ func (c *Controller) doHealthCheck(cancel <-chan struct{}, key health.HealthStat
...
+		ProxyPort:     5601,
+		HostIP:        hostip,
+		Protocol:      "tcp",
+	}
+	endpoints["tcp:5601"] = []applicationendpoint.ApplicationEndpoint{kibana_endpoint}
+
+	return endpoints
+}
+
 func (c *Controller) handleControlCenterImports(rpcdead chan struct{}) error {
```
Comment (tgjonestx) [inline container/controller.go]: Please remove `rpcdead` just because it's cruft at this point

File: container/controller.go [hunk 2]
```diff
@@ -790,103 +790,106 @@ func (c *Controller) doHealthCheck(cancel <-chan struct{}, key health.HealthStat
...
+		},
+		PrivateIP:  endpoint.ContainerIP,
+		HostIP:     endpoint.HostIP,
+		InstanceID: 0,
+	}
+	c.endpoints.UpdateRemoteExports(bind, exports)
+	return
+}
+
+func (c *Controller) createControlplaneEndpoints() map[string][]applicationendpoint.ApplicationEndpoint {
```
Comment (smousa) [inline container/controller.go]: Why are you doing this create when you are going to have to convert it anyway?

File: node/agent_proxy.go [hunk 3]
```diff
@@ -54,21 +52,9 @@ func (a *HostAgent) Ping(waitFor time.Duration, timestamp *time.Time) error {
...
-	a.addKibanaEndpoint(myList)
-
-	*response = myList
-	return nil
-}
-
 func (a *HostAgent) GetEvaluatedService(request EvaluateServiceRequest, response *EvaluateServiceResponse) (err error) {
 	logger := plog.WithFields(log.Fields{
-		"serviceID": request.ServiceID,
+		"serviceID":  request.ServiceID,
```
Comment (smousa) [inline node/agent_proxy.go]: should be all lowercase

File: container/controller.go (hunk 2 above)
Comment (ed-) [inline container/controller.go]: I'll move that out to its own spot and just create the registry.ExportDetails that they get turned into, like we just discussed.

File: node/agent_proxy.go (hunk 3 above)
Comment (ed-) [inline node/agent_proxy.go]: Done

File: container/controller.go (hunk 1 above)
Comment (ed-) [inline container/controller.go]: Done

File: container/cc_endpoints.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import (
+	"github.com/control-center/serviced/zzk/registry"
+	zkservice "github.com/control-center/serviced/zzk/service"
+
+	"os"
+	"strings"
+)
+
+var (
+	master_ip = os.Getenv("SERVICED_MASTER_IP")
```
Comment (smousa) [inline container/cc_endpoints.go]: There is a new way to get config options from serviced.  Check with @iancmcc

File: container/cc_endpoints.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	hostip    = strings.Split(master_ip, ":")[0]
+
+	cp_controlplane_bind = zkservice.ImportBinding{
```
Comment (smousa) [inline container/cc_endpoints.go]: snake case is faux pas in go, please switch to camel

File: node/agent_proxy.go [hunk 6]
```diff
@@ -54,7 +54,7 @@ func (a *HostAgent) Ping(waitFor time.Duration, timestamp *time.Time) error {
 
 func (a *HostAgent) GetEvaluatedService(request EvaluateServiceRequest, response *EvaluateServiceResponse) (err error) {
 	logger := plog.WithFields(log.Fields{
-		"serviceID":  request.ServiceID,
+		"serviceid":  request.ServiceID,
 		"instanceID": request.InstanceID,
```
Comment (smousa) [inline node/agent_proxy.go]: can you change instanceID to instanceid as well?

**Pull request closed without merging**